envi/archs/arm/disasm.py

```diff
@@ -2083,27 +2140,30 @@ def getOperAddr(self, op, emu=None):
         # there are certain circumstances where we can survive without an emulator
         pubwl = self.pubwl >> 3
         u = pubwl & 1
-
+        tsize = self.tsize # to help cope with ldcl
         # if we don't have an emulator, we must be PC-based since we know it
         if self.base_reg == REG_PC:
             base = self.va
+            #to handle lcdl. Override tsize to return proper address
+            if ((self.pubwl >>2) & 1) == 1:
+                tsize =4
         elif emu == None:
             return None
         else:
             base = emu.getRegister(self.base_reg)
 
         if u:
-            addr = (base + self.offset) & e_bits.u_maxes[self.tsize]
+            addr = (base + self.offset) & e_bits.u_maxes[tsize]
```

Comment:
this should be different as of my latest updates.  please pull them in.  you may have to resolve conflicts.  the new/good version should use "self.psize".